install_ads

```diff
@@ -0,0 +1,712 @@
+    if [[ "$OS" == "osx" ]]; then
+      while [[ ! "$GCC_INSTALL" =~ ^([Yy](es)?|[Nn]o?)$ ]]; do
+        GCC_INSTALL="no"
+        if [ -z "$NONINTERACTIVE" ]; then
+          print question "Do you want to install it now? (accepting to install \"gcc\" will also install \"brew\" in case it is not installed already)? [y]es/[n]o: "
+          read -r GCC_INSTALL
+        fi
+      done
+      if [[ "$GCC_INSTALL" =~ [Yy](es)? ]]; then
+        if [[ ! $(command -v brew) ]]; then
+          # NB: this is a different NONINTERACTIVE than ours above; it's for the brew installer
+          (NONINTERACTIVE=1 /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install.sh)")
+        fi
+        brew install -f gcc
+        # check if gcc install ran properly
+        if ! gcc --version > /dev/null 2>&1; then
+          die "ERROR: Installation of \"gcc\" failed. Please contact ADS team for assistance."
+        fi
+      else
+        die "Please install \"gcc\" and restart ADS installation."
+      fi
+    else
+      die "Please install \"gcc\" and restart ADS installation. On Debian/Ubuntu, run: \"apt install gcc\". On CentOS/RedHat, run: \"yum -y install gcc\"."
+    fi
+  fi
+  print info "OK!"
+}
+
+# Gets the shell rc file path based on the default shell.
+# @output: THE_RC and RC_FILE_PATH vars are modified
+function get_shell_rc_path() {
+  if [[ "$SHELL" == *"bash"* ]]; then
+    THE_RC="bash"
+    RC_FILE_PATH="$HOME/.bashrc"
+  elif [[ "$SHELL" == *"/sh"* ]]; then
+    THE_RC="bash"
+    RC_FILE_PATH="$HOME/.bashrc"
+  elif [[ "$SHELL" == *"zsh"* ]]; then
+    THE_RC="bash"
+    RC_FILE_PATH="$HOME/.zshrc"
+  elif [[ "$SHELL" == *"csh"* ]]; then
+    THE_RC="csh"
+    RC_FILE_PATH="$HOME/.cshrc"
+  else
+    find ~/.* -maxdepth 0 -type f
+    die "ERROR: Shell was not recognized: $SHELL"
+  fi
+}
+
+# Force bashrc loading
+function force_bashrc_loading() {
+  sourceblock="
+if [[ -n \"\$BASH_VERSION\" ]]; then
+    # include .bashrc if it exists
+    if [[ -f \"\$HOME/.bashrc\" ]]; then
+    . \"\$HOME/.bashrc\"
+    fi
+fi"
+  bidon=""
+  for profile in ~/.bash_profile ~/.bash_login ~/.profile; do
+    if [[ -a "$profile" ]]; then
+      if ! grep -E "(\.|source) .*bashrc" "$profile" >/dev/null 2>&1; then
+        echo "$sourceblock" >>"$profile"
+      fi
+      bidon="done"
+      break
+    fi
+  done
+
+  if [[ -z "$bidon" ]]; then
+    echo "$sourceblock" >>~/.bash_profile
+  fi
+}
+
+# Installation text to insert in shell config file
+function edit_shellrc() {
+  # Write text common to all shells
+  (
+    echo
+    echo ""
+    echo "# AXONDEEPSEG (installed on $(date +%Y-%m-%d\ %H:%M:%S))"
+    echo "$DISPLAY_UPDATE_PATH"
+    # Switch between shell
+    if [[ "$THE_RC" == "bash" ]]; then
+      echo "export ADS_DIR=$ADS_DIR"
+    elif [[ "$THE_RC" == "csh" ]]; then
+      echo "setenv ADS_DIR $ADS_DIR"
+    fi
+    # add line
+    echo ""
+  ) >> "$RC_FILE_PATH"
+}
```

Comment:
Note that one of the bigger changes in this script is how the CLI scripts are accessed:

- Before: Necessary to activate the conda environment
- After: Specific launchers (`axondeepseg_*`, `download_*`) are copied to a separate `bin/` folder, which is then made available via the `.{bash,zsh}rc` file in the user's home directory.

There are some caveats with this approach, namely the pollution of said `.rc` file with repeated installs + it is tricky to have multiple concurrent versions (as the most recently installed version supersedes the other).

I'd love to chat about the caveats of SCT's installation procedure, as it could retroactively help SCT as well. :)